SCloudRandomizer/MainViewController.m

```diff
@@ -373,17 +378,31 @@ - (void)remoteControlReceivedWithEvent:(UIEvent *)event
     }
 }
 
+// Do something based on specific notifications
+- (void)receiveNotification:(NSNotification *) notification {
+    if ([[notification name] isEqualToString:@"StreamCompleted"]) {
+        NSLog (@"Recieved StreamCompleted notification!");
+        [self playNextTrack];
+    } else if ([[notification name] isEqualToString:@"ReadyToPlay"]) {
+        NSLog(@"Recieved ReadyToPlay notification!");
+        if (self.prepToPlayHud != nil) {
```

Comment:
can self.prepToPlayHud be null here? Again we should just fail hard to catch these cases I think